cunumeric/eager.py

```diff
@@ -431,6 +456,57 @@ def diag(self, rhs, extract, k, stacklevel):
             self.array[:] = np.diag(rhs.array, k)
             self.runtime.profile_callsite(stacklevel + 1, False)
 
+    def diag_helper(
+        self,
+        rhs,
+        diag_size,
+        offset,
+        naxes,
+        extract,
+        stacklevel=0,
+        callsite=None,
+    ):
+        if self.shadow:
+            rhs = self.runtime.to_eager_array(rhs, stacklevel=(stacklevel + 1))
+        elif self.deferred is None:
+            self.check_eager_args((stacklevel + 1), rhs)
+        if self.deferred is not None:
+            self.deferred.diag_helper(
+                rhs,
+                diag_size,
+                offset,
+                naxes,
+                extract,
+                stacklevel=(stacklevel + 1),
+            )
+        else:
+            # this should be called only if number of axes <=2
+            if naxes == 2:
+                ndims = rhs.array.ndim
+                self.array[:] = np.diagonal(
+                    rhs.array, offset=offset, axis1=ndims - 2, axis2=ndims - 1
+                )
+            elif naxes < 2:
```

Comment:
Shouldn't this just throw an error?

Reply:
This has already been checked in array.py, I would just `assert naxes > 2`

Reply:
Actually, `diagonal_helper` is used for both `diag` and `diagonal`. So I believe that adding `assert` would be incorrect

Reply:
You're right, I got confused.